File: solutions/r/bob/1/bob.R
is_yelling <- function(input) {
  grepl("[[:upper:]]", input) & input == toupper(input)
}

is_question <- function(input) {
  grepl("\\?$", input)
}

bob <- function(input) {
  trimmed <- gsub("^\\s+|\\s+$", "", input)
  if (nchar(trimmed) == 0) {
    "Fine. Be that way!"
  } else if (is_question(trimmed)) {
    if (is_yelling(trimmed)) {
      "Calm down, I know what I'm doing!"
    } else {
      "Sure."
    }
  } else if (is_yelling(trimmed)) {
    "Whoa, chill out!"
  } else {
    "Whatever."
  }
}

File: solutions/python/bob/1/bob.py
import re

def response(hey_bob):
    if hey_bob.strip() == "":
        return "Fine. Be that way!"
    if hey_bob.upper() == hey_bob and re.search(r"[A-Z]", hey_bob):
        if hey_bob.strip().endswith("?"):
            return "Calm down, I know what I'm doing!"
        return "Whoa, chill out!"
    if hey_bob.strip().endswith("?"):
        return "Sure."
    return "Whatever."
